packages/ckeditor5-emoji/src/emojipicker.ts

```diff
@@ -0,0 +1,191 @@
+/**
+ * @license Copyright (c) 2003-2024, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module emoji/emojipicker
+ */
+
+import { Picker } from 'emoji-picker-element';
+
+import {
+	ButtonView,
+	clickOutsideHandler,
+	ContextualBalloon,
+	View
+} from 'ckeditor5/src/ui.js';
+
+import { Plugin, type Editor } from 'ckeditor5/src/core.js';
+import EmojiLibraryIntegration from './emojilibraryintegration.js';
+
+type EmojiClickEvent = Event & {
+	detail: {
+		unicode: string;
+	};
+};
+
+/**
+ * The emoji picker plugin.
+ *
+ * Introduces the `'emoji'` dropdown.
+ */
+export default class EmojiPicker extends Plugin {
+	private _balloon!: ContextualBalloon;
+	declare private _emojiDialog: EmojiDialog;
+
+	/**
+	 * @inheritDoc
+	 */
+	public static get requires() {
+		return [ EmojiLibraryIntegration, ContextualBalloon ] as const;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public static get pluginName() {
+		return 'EmojiPicker' as const;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public static override get isOfficialPlugin(): true {
+		return true;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public init(): void {
+		this._balloon = this.editor.plugins.get( ContextualBalloon );
+		this._emojiDialog = this._createEmojiDialog();
+
+		this.editor.ui.componentFactory.add( 'emoji', () => {
```

Comment:
We need a dedicated integration with the menu bar, right? Example: https://github.com/ckeditor/ckeditor5/blob/db95303a82cb0aa649bc2d16920b71f531c1fc66/packages/ckeditor5-basic-styles/src/bold/boldui.ts#L51

Reply:
It would also require a change in the menu bar configuration itself.